### PR TITLE
fix: schema preview data ui changed for sql, postgres, mongo

### LINKED ISSUE #27946: [Task]: Update the sql datasource preview data UI
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Update the UI for datasource preview data as per new designs
- Convert the above components as common for supporting gsheet preview data UI as well.

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
This PR includes the changes :
- Revamp implementation for the display of tabs on datasource view mode for sql, postgres, mongo
- Update the UI for the tabs on the datasource view mode according to the updated design

figma link : https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=3187-16095&mode=dev
notion link : https://www.notion.so/appsmith/5c8aa998aa94429995a6e0825ca2df05?v=14883d075b6649be9a8748938dcc224b&p=b86311e7fbd34d1ab5a2a90af2167518&pm=s

#### PR fixes following issue(s)
Fixes #27946 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by chandannkumar** [comment]
@AmanAgarwal041 
Observation:
- The horizontal scroll is not visible until page is scrolled to bottom. (this is because In prod, we have 2 vertical scroll and here we have 1).
https://www.loom.com/share/e2c12a1b15a64995a3ef7245a26f067c?sid=208556d2-1d85-4c1c-81ed-40fe93d710de

- Can we show the separator in column header so that user can adjust the breadth of a particular column? As it allows user to resize the column header
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXhYQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1b2d6b08dba7a896a71a83db78eca8f5d024d0d4/image.png)

**Comment by rohan-arthur** [comment]
@AmanAgarwal041 @chandannkumar  I have created a [separate card](https://www.notion.so/appsmith/5c8aa998aa94429995a6e0825ca2df05?v=14883d075b6649be9a8748938dcc224b&p=247103a7082f4eebaedc00b036c01d08&pm=s) to track the resizer question, since we need a decision here. 

Let's proceed with the review/qa for the rest.

**Comment by chandannkumar** [comment]
Tested this PR and LGTM
- Tested for Postgres, MySQL, Mongo(only mock db)
- Observed and verified design changes

File: app/client/src/pages/Editor/DataSourceEditor/index.tsx [hunk 1]
```diff
@@ -846,6 +855,46 @@ class DatasourceEditorRouter extends React.Component<Props, State> {
...
+  shouldShowTabs = () => {
+    const {
+      isEnabledForDSViewModeSchema,
+      isPluginAllowedToPreviewData,
+      pluginDatasourceForm,
+    } = this.props;
+    const isRestAPI =
+      pluginDatasourceForm === DatasourceComponentTypes.RestAPIDatasourceForm;
+    return (
+      isEnabledForDSViewModeSchema && isPluginAllowedToPreviewData && !isRestAPI
```
Comment (sneha122) [inline app/client/src/pages/Editor/DataSourceEditor/index.tsx:882]: why are we checking for isRestAPI explicitly? Anyways `isPluginAllowedToPreviewData` should be false for restAPI right?

File: app/client/src/pages/Editor/DatasourceInfo/DatasourceViewModeSchema.tsx [hunk 2]
```diff
@@ -303,67 +302,67 @@ const DatasourceViewModeSchema = (props: Props) => {
...
+              step={0}
+              tableName={tableName}
+            />
+          </DatasourceListContainer>
+        </StructureContainer>
+        <DatasourceDataContainer>
+          <TableWrapper>
+            {isLoading && (
+              <MessageWrapper>
+                <Spinner size="md" />
```
Comment (sneha122) [inline app/client/src/pages/Editor/DatasourceInfo/DatasourceViewModeSchema.tsx:327]: Figma design has a different loading state, and over here we are using spinner, is this expected

File: app/client/src/pages/Editor/DatasourceInfo/DatasourceViewModeSchema.tsx [hunk 3]
```diff
@@ -303,67 +302,67 @@ const DatasourceViewModeSchema = (props: Props) => {
...
+                <Text style={{ marginLeft: "8px" }}>
+                  {createMessage(FETCHING_DATASOURCE_PREVIEW_DATA)}
+                </Text>
+              </MessageWrapper>
+            )}
+            {!isLoading && failedFetchingPreviewData && (
+              <MessageWrapper>
```
Comment (sneha122) [inline app/client/src/pages/Editor/DatasourceInfo/DatasourceViewModeSchema.tsx:334]: Same here as well, failed state is shown using an image and a retry CTA in designs, is this expected?

File: app/client/src/pages/Editor/DataSourceEditor/index.tsx (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/DataSourceEditor/index.tsx:882]: Yes make sense. Will remove this check in https://github.com/appsmithorg/appsmith/pull/28318

File: app/client/src/pages/Editor/DatasourceInfo/DatasourceViewModeSchema.tsx (hunk 2 above)
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/DatasourceInfo/DatasourceViewModeSchema.tsx:327]: These error states will be handled in the issue : https://www.notion.so/appsmith/Loading-and-error-state-for-schema-1f99652227ad4eada949fa72334154d8

**Review by sneha122 — APPROVED** [review]

**Pull request merged**